pkg/kopia/command/kopia.go

```diff
@@ -59,3 +71,188 @@ func commonArgs(password, configFilePath, logDirectory string, requireInfoLevel
+		args := logsafe.Cmd{}
+		args = args.AppendRedactedKV(accessKeyFlag, prof.AccessKeyID())
+		args = args.AppendRedactedKV(secretAccessKeyFlag, prof.SecretAccessKey())
+		return args, nil
+	default:
+		return nil, errors.New(fmt.Sprintf("Unsupported type for credentials %s", credsType))
+	}
+}
+
+func kopiaAzureArgs(prof kopia.Profile, artifactPrefix string) (logsafe.Cmd, error) {
+	artifactPrefix = GenerateFullRepoPath(prof.Prefix(), artifactPrefix)
+
+	args := logsafe.NewLoggable(azureSubCommand)
+	args = args.AppendLoggableKV(containerFlag, prof.BucketName())
+	args = args.AppendLoggableKV(prefixFlag, artifactPrefix)
+
+	credArgs, err := kopiaAzureCredentialArgs(prof)
+	if err != nil {
+		return nil, err
+	}
+
+	return args.Combine(credArgs), nil
+}
+
+func kopiaAzureCredentialArgs(prof kopia.Profile) (logsafe.Cmd, error) {
+	credsType, err := prof.CredType()
+	if err != nil {
+		return nil, err
+	}
+	var storageAccount, storageKey, storageEnv string
+	switch credsType {
+	case kopia.SecretTypeK8sSecret:
+		azureSecret, err := secrets.ExtractAzureCredentials(prof.Secret())
+		if err != nil {
+			return nil, err
+		}
+		storageAccount = azureSecret.StorageAccount
+		storageKey = azureSecret.StorageKey
+		storageEnv = azureSecret.EnvironmentName
+	case kopia.SecretTypeKeyPair:
+		storageAccount = prof.StorageAccount()
+		storageKey = prof.StorageKey()
+		storageEnv = prof.StorageDomain()
+	default:
+		return nil, errors.New(fmt.Sprintf("Unsupported type for credentials %s", credsType))
+	}
+
+	args := logsafe.Cmd{}
+	args = args.AppendRedactedKV(storageAccountFlag, storageAccount)
+	args = args.AppendRedactedKV(storageKeyFlag, storageKey)
+	if storageEnv != "" {
+		env, err := azure.EnvironmentFromName(storageEnv)
+		if err != nil {
+			return nil, errors.Wrap(err, fmt.Sprintf("Failed to get azure environment from name %s", storageEnv))
+		}
+		blobDomain := "blob." + env.StorageEndpointSuffix
+		args = args.AppendLoggableKV(storageDomainFlag, blobDomain)
+	}
+	return args, nil
+}
+
+func kopiaGCSArgs(prof kopia.Profile, artifactPrefix string) logsafe.Cmd {
+	artifactPrefix = GenerateFullRepoPath(prof.Prefix(), artifactPrefix)
+
+	args := logsafe.NewLoggable(googleSubCommand)
+	args = args.AppendLoggableKV(bucketFlag, prof.BucketName())
+	args = args.AppendLoggableKV(credentialsFileFlag, consts.GoogleCloudCredsFilePath)
+	return args.AppendLoggableKV(prefixFlag, artifactPrefix)
+}
+
+func filesystemArgs(prof kopia.Profile, artifactPrefix string) logsafe.Cmd {
+	artifactPrefix = GenerateFullRepoPath(prof.Prefix(), artifactPrefix)
+
+	args := logsafe.NewLoggable(filesystemSubCommand)
+	return args.AppendLoggableKV(pathFlag, kopia.DefaultFSMountPath+"/"+artifactPrefix)
+}
+
+func kopiaBlobStoreArgs(prof kopia.Profile, artifactPrefix string) (logsafe.Cmd, error) {
+	locType, err := prof.LocationType()
+	if err != nil {
+		return nil, err
+	}
+	switch locType {
+	//case LocationTypeFileStore:
+	//	return filesystemArgs(prof, artifactPrefix), nil
```

Comment:
Needs cleanup?

Reply:
I kept it purposely so that it grabs reviewer attention. Thanks for noticing... 😅 I remember @pavannd1 mentioned a while back that we should have some one work on adding filestore support to Kanister profiles separately.